src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowHistorySpec.groovy

```diff
@@ -0,0 +1,106 @@
+package org.openkilda.functionaltests.spec.flows.haflows
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.helpers.HaFlowHelper
+import org.openkilda.northbound.dto.v2.haflows.HaFlow
+import org.openkilda.testing.service.northbound.model.HaFlowActionType
+
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Narrative("""Verify that history records are created for the basic actions applied to Ha-Flow.""")
+class HaFlowHistorySpec extends HealthCheckSpecification {
+
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Shared
+    boolean isDeleted
+
+    @Tidy
+    def "User can change an Ha-Flow and get its history event - #type"() {
+        given: "HA-Flow"
+        def swT = topologyHelper.switchTriplets[0]
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT)
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+
+        when: "Change the flow"
+        change(haFlow)
+
+        then: "Correct event appears in HA-Flow history"
+        haFlowHelper.getHistory(haFlow.haFlowId).hasExactlyNEntriesOfType(type, 1)
+
+        cleanup:
+        !isDeleted && haFlowHelper.deleteHaFlow(haFlow.haFlowId)
```

Comment:
typo: missed haFlow in the check 
cleanup:
        haFlow && !isDeleted && haFlowHelper.deleteHaFlow(haFlow.haFlowId)